source/common/secret/secret_manager_impl.cc

```diff
@@ -79,5 +84,61 @@ SecretManagerImpl::findOrCreateCertificateValidationContextProvider(
                                                     secret_provider_context);
 }
 
+ProtobufTypes::MessagePtr SecretManagerImpl::dumpSecretConfigs() {
+  auto config_dump = std::make_unique<envoy::admin::v2alpha::SecretsConfigDump>();
+  auto providers = certificate_providers_.allSecretProviders();
+  for (const auto& cert_secrets : providers) {
+    const auto& secret_data = cert_secrets->secretData();
+    const auto& tls_cert = cert_secrets->secret();
+    ::envoy::admin::v2alpha::SecretsConfigDump_DynamicSecret* dump_secret;
+    bool secret_ready = tls_cert != nullptr;
+    if (secret_ready) {
+      dump_secret = config_dump->mutable_dynamic_active_secrets()->Add();
+    } else {
+      dump_secret = config_dump->mutable_dynamic_warming_secrets()->Add();
+    }
+    auto secret = dump_secret->mutable_secret();
+
+    ProtobufWkt::Timestamp last_updated_ts;
+    TimestampUtil::systemClockToTimestamp(secret_data.last_updated_, last_updated_ts);
+    dump_secret->set_version_info(secret_data.version_info_);
+    *dump_secret->mutable_last_updated() = last_updated_ts;
+    secret->set_name(secret_data.resource_name);
+    if (secret_ready) {
+      auto tls_certificate = secret->mutable_tls_certificate();
+      tls_certificate->MergeFrom(*tls_cert);
+      // We clear private key and password to avoid information leaking.j
```

Comment:
Nit: accidental `j` at the end of this line.

Reply:
done.